lib/dry/files.rb

```diff
@@ -64,14 +69,16 @@ def touch(path)
     # All the intermediate directories are created.
     #
     # @param path [String,Pathname] the path to file
-    # @param content [String, Array<String>] the content to write
+    # @param lines [String, Array<String>] the content to write
     #
     # @raise [Dry::Files::IOError] in case of I/O error
     #
     # @since 0.1.0
     # @api public
-    def write(path, *content)
-      adapter.write(path, *content)
+    def write(path, lines)
+      joined_lines = Array(lines).flatten.map { |line| line.chomp.concat(newline) }.join
+      joined_lines = "" if joined_lines == "\n" # Leave it empty
```

Comment:
@cllns Can we extract `""` as `EMPTY_STRING.dup`?
Also, why `"\n"` is hardcoded, given we have `@newline`?